shared/chaincode_tools/update_chaincodes.py

```diff
@@ -11,38 +11,42 @@
 import subprocess
 import re
 from multiprocessing.pool import ThreadPool
-
-DEBUG = False
-GOPATH = os.environ['GOPATH']
-BASE_CONF_PATH = GOPATH + '/src'
-
-DRYRUN = False # Debug only
-if DRYRUN:
-    BASE_CONF_PATH = GOPATH
-
-CONF_FILE = BASE_CONF_PATH + '/chaincodes.json'
+from argparse import ArgumentParser
+from argparse import RawTextHelpFormatter
+
+PARSER = ArgumentParser(description="""Instantiates new chaincodes or upgrades existing ones.
+Chaincodes should be in $GOPATH/src/chaincodes (optional for nodejs, see below). Supports nodejs and go chaincodes.
+The script looks for a file $GOPATH/src/chaincodes.json which must contain the paths to the chaincodes, relative to src/chaincodes/
+
+* For nodejs chaincode only:
+    > Compiled Nodejs chaincodes will be in $GOPATH/src/build.
+    > If you provide a repository, this will pull from it and save it in $GOPATH/src/, and then run `npm run build`, which must create the $GOPATH/src/build folder, containing chaincodes.
+    > If no repository is given, the script will only look in $GOPATH/src/build, making the contents of $GOPATH/src/chaincodes optional.
+""", formatter_class=RawTextHelpFormatter)
+PARSER.add_argument('--dryrun', help='Shows which commands would be run, without running them', action='store_true')
+PARSER.add_argument('--repository', '-r', type=str,help='the repository from which the chaincode should be fetched. If not given, assumes chaincodes are in $GOPATH/src/build/')
+PARSER.add_argument('--forceRebuild', '-f', help='forces the script to run npm install on each chaincode. By default it will only run npm install when the node_modules directory for that chaincode is missing', action='store_true')
```

Comment:
not sure if foreRebuild is a good name here ... this might conflict with the build step

forceNpmInstall or something like that or foreInstallDeps ...